casper/src/main/scala/coop/rchain/casper/blocks/proposer/Proposer.scala

```diff
@@ -37,24 +50,22 @@ object ProposerResult {
   def started(seqNumber: Long): ProposerResult = ProposerStarted(seqNumber)
 }
 
-class Proposer[F[_]: Concurrent: Log: Span](
-    getLatestSeqNumber: Validator => F[Long],
-    // propose constraint checkers
+final case class Proposer[F[_]: Concurrent: Log: Span: RuntimeManager: BlockStore: BlockDagStorage](
     checkActiveValidator: ValidatorIdentity => F[Boolean],
-    createBlock: ValidatorIdentity => F[BlockCreatorResult],
+    createBlock: (ParentsMergedState, ValidatorIdentity) => F[BlockCreatorResult],
     validateBlock: BlockMessage => F[ValidBlockProcessing],
     proposeEffect: BlockMessage => F[Unit],
     validator: ValidatorIdentity
```

Comment:
Why to have all these methods accepted separately? Proposer can just work with instance like BlockCreator to initiate block creator execution.
Take a look how easily can be tested code with dependencies with Mockito library.
- https://github.com/rchain/rchain/pull/3787

We now have 3 different things about Proposer which is confusing. For example `ProposerInstance` doesn't create Proposer instance, it just accepting it as an argument.

I'd like to see clear hierarchy of dependencies. BlockCreator can have methods with the logic to prepare each step in block creation process and Proposer can be the _worker_ how is responsible to call the final method from BlockCreator in the right moment and/or from the right place.